<?xml version="1.0"?>
<launch>
    <arg name="paused" default="true"/>
    <arg name="uwsim" default="false"/>

    <!-- Launch Gazebo with underwater world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="false" if="$(arg uwsim)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="world_name" value="$(find freefloating_gazebo)/world/underwater.world"/>
    </include>
  
    <!-- include gazebo spawner (generated while parsing uwsim launcher) -->
    <group ns="ecn_auv">
        <param name="robot_description" command="$(find xacro)/xacro.py $(find ecn_rasom)/urdf/ecn_auv.xacro"/>
        <node name="spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-urdf -model ecn_auv -param robot_description -x 0 -y 0 -z 1 -R 0 -P 0 -Y 0"/>

        <!-- Load controller configurations (vehicle and arm) from YAML file to parameter server -->
        <rosparam file="$(find ecn_sonar)/config/ecn_auv_control.yaml" command="load"/>
  
        <!-- Launch low-level PID control for body and arm -->
        <node name="pid_control" pkg="freefloating_gazebo" type="pid_control" output="screen"/>
    </group>
    
    
      <group>
    <param name="terrain_description" command="$(find xacro)/xacro.py $(find ecn_sonar)/urdf/terrain.sdf"/>
    <node name="terrain_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-sdf -model terrain -param terrain_description -x 0 -y 0 -z 0 -R 0 -P 0 -Y 0"/>
  </group>
  
</launch>
